chain/client/src/client_actor.rs

```diff
@@ -121,8 +123,18 @@ impl ClientActor {
         validator_signer: Option<Arc<dyn ValidatorSigner>>,
         telemetry_actor: Addr<TelemetryActor>,
         enable_doomslug: bool,
+        ctx: &Context<ClientActor>,
         #[cfg(feature = "adversarial")] adv: Arc<RwLock<AdversarialControls>>,
     ) -> Result<Self, Error> {
+        let state_parts_arbiter = Arbiter::new();
```

Comment:
An `Arbiter` is basically an `std::thread`, so the two usual questions apply: 

* how do we make sure that the thread eventually stops? 
* how do we make sure that, when the thread panics due to a bug, we eventually notice this and tear-down the process? 

For the first, I think what we might want to do is to store the arbiter as a field of `ClientActor`, and call `arbiter.stop(); arbiter.join()` in the `Drop` impl. I think we should do that, otherwise writing tests becomes cumbersome, as `StatePartsActor` from old test might somehow interferer with newer tests. 

Not sure what we need to do with the second case. What would be behavior when `apply_state_part` panics? I guess the client actor will be just infinitely stuck in the old epoch. Seems that, if the background actor dies, the `do_send` will just drop messages on the floor. Is this an OK behavior? Can we link the ClientActor and the StatePartsActor such that the client is guaranteed to die after `StatePartsActor`?

Reply:
> Can we link the ClientActor and the StatePartsActor such that the client is guaranteed to die after StatePartsActor?

That is an interesting question. I am not sure whether that is necessarily something we want to do. It sort of depends on why exactly `apply_state_part` panicked and we could potentially still recover by retrying.